:sparkles: support spring-boot  4.0.0、spring-cloud 2025.1.0

xuxiaowei-com-cn opened the pull request:
### Describe what this PR does / why we need it

https://github.com/alibaba/spring-cloud-alibaba/issues/4067

### Does this pull request fix one issue?

<!--If that, add "Fixes #xxxx" below in the next line. For example, Fixes #15. Otherwise, add "NONE" -->

### Describe how you did it


### Describe how to verify it


### Special notes for reviews

xuxiaowei-com-cn committed:
:sparkles: support spring-cloud 2025.1.0

https://github.com/alibaba/spring-cloud-alibaba/issues/4067

xuxiaowei-com-cn committed:
:art: checkstyle

xuxiaowei-com-cn commented:
@uuuyuqi 请问一下，如何解决这个代码格式的问题？

1. 如果我将 `Copyright 2013-2023 the original author or authors.` 替换为 `Copyright 2013-present the original author or authors.`  https://github.com/xuxiaowei-com-cn/spring-cloud-alibaba/commit/752d367c9a92a73dfffe8c30176d22c2442d7fb4 （修改了 452 个文件） 确实可以解决此问题，但是修改的文件数量太多了。
2. 如果不修改 `Copyright 2013-2023 the original author or authors.` 则会出现问题：https://github.com/alibaba/spring-cloud-alibaba/actions/runs/18028214112
3. 如果有其他方案，我会重新整理提交的代码。

<img width="1956" height="1008" alt="image" src="https://github.com/user-attachments/assets/9c716917-047c-4eef-be79-c4be2bb5cea2" />

uuuyuqi commented:
你好，2013-present 是更加合理的。我会尽快提交一个 PR 先对这些内容统一进行修改，方便后续大家提PR。

xuxiaowei-com-cn commented:
@uuuyuqi 好的，谢谢，到时候我再重新处理一下这个 PR

uuuyuqi commented:
我仔细看了下这个 style check 报错，目前看下来是 SpringCloud 社区从 2025.0.0 演进到 2025.1.0 时，修改了 [spring-cloud-build-tools/src/main/resources/checkstyle-header.txt ](https://github.com/spring-cloud/spring-cloud-build/blob/v5.0.0-M2/spring-cloud-build-tools/src/main/resources/checkstyle-header.txt)校验内容，从 `Copyright \E20\d\d\-20\d\d\Q` 改为了 `Copyright \E20\d\d\-present\Q`。SpringCloudAlibaba 升级到 SpringCloud 2025.1.x 时也会升级 spring-cloud-build-tools 版本，也就会出现这个 style check 报错，感谢 @xuxiaowei-com-cn 的发现！后续 SpringCloudAlibaba 社区会 follow 这个变更。 

不过目前 SpringCloud 2025.1.x 尚未正式发布，SpringCloudAlibaba 也会优先发布 2025.0.x 版本，大概率就10月份，具体时间暂时无法确定。cc @LearningGp 

这个 PR 的审查和合并会先搁置，等到 2025.0.0 发布后会继续启动，到时候我们一起讨论下这里的 style 如何修改更合适~

xuxiaowei-com-cn commented:
@uuuyuqi OK

xuxiaowei-com-cn commented:
@uuuyuqi 前两天已经重新整理此 pr

xuxiaowei-com-cn committed:
:arrow_up: spring-boot upgrade from 4.0.0-M3 to 4.0.0-RC1, Disabled FeignClientCircuitBreakerRuleIntegrationTest

xuxiaowei-com-cn committed:
:bug: fix WebClientCustomizer package

xuxiaowei-com-cn committed:
:art: checkstyle

xuxiaowei-com-cn committed:
:arrow_up: spring-boot upgrade from 4.0.0-RC1 to 4.0.0-RC2

xuxiaowei-com-cn committed:
:arrow_up: spring-cloud upgrade from 2025.1.0-M4 to 2025.1.0-RC1

xuxiaowei-com-cn committed:
:arrow_up: spring-boot upgrade from 4.0.0-RC2 to 4.0.0

1. disable RocketMQMessageChannelBinderTest
2. enable FeignClientCircuitBreakerRuleIntegrationTest

xuxiaowei-com-cn committed:
:arrow_up: spring-cloud upgrade from 2025.1.0-RC1 to 2025.1.0

1. internal classes use static final modifier
2. fix nacos instance url
3. fix NacosAutoServiceRegistration Bean
4. fix NacosDiscoveryLoadBalancerConfigurationTest

https://github.com/alibaba/spring-cloud-alibaba/issues/4067

xuxiaowei-com-cn commented:
@uuuyuqi @LearningGp 您好，spring-boot 4.0.0、spring-cloud 2025.1.0 已经发布正式版，麻烦您有时间处理一下。

kuangyoubo commented:
@xuxiaowei-com-cn 2025.1.0 预计什么时候可以发布

xuxiaowei-com-cn commented:
@kuangyoubo 我只是项目一个普通贡献者，并没有此项目的管理权限。
如果你着急用，可以参考 https://github.com/alibaba/spring-cloud-alibaba/issues/4067#issuecomment-3333455280 ，使用我自己编译的版本。

LearningGp commented:
社区还在收集相关的问题，修复这些问题后会先发布2025.0.0.1，然后再处理 2025.1.0.0 版本，这两个版本目前都预计在12月份完成

xuxiaowei-com-cn committed:
:bug: fix EnvironmentPostProcessorApplicationListener package name

shaobin0613 commented:
> 社区还在收集相关的问题，修复这些问题后会先发布2025.0.0.1，然后再处理 2025.1.0.0 版本，这两个版本目前都预计在12月份完成

spring boot 4升级了很多东西，想尝试下spring boot的RequestMapping version，期待2025.0.0.1早日发布

kuangyoubo commented:
> @kuangyoubo 我只是项目一个普通贡献者，并没有此项目的管理权限。 如果你着急用，可以参考 [#4067 (comment)](https://github.com/alibaba/spring-cloud-alibaba/issues/4067#issuecomment-3333455280) ，使用我自己编译的版本。

谢谢，先用你的方式适配到。我在做老项目迁移，结果在这里卡住了，哈哈哈

lcxw commented:
please add support for spring-cloud-gateway-server-mvc which benefit from virtual threads. @xuxiaowei-com-cn ❤️

1.  spring-cloud-gateway-server-mvc for spring-cloud-alibaba with  nacos config  support

lcxw commented:
https://github.com/alibaba/spring-cloud-alibaba/issues/3916

xuxiaowei-com-cn commented:
@lcxw 

1. spring-cloud-starter-gateway-server-webflux 是 webflux
2. spring-cloud-starter-gateway-server-webmvc 是 mvc
3. spring-cloud-alibaba 之前只支持 gateway webflux
4. 如果要添加 gateway mvc，需要增加一个 Maven 模块（坐标）
5. https://github.com/alibaba/spring-cloud-alibaba/issues/3916 议题很久没有处理，并且据我观察 gateway mvc 并不受重视
6. 而且 gateway 不建议使用 mvc，不如 webflux 性能高
7. 如果要增加一个 Maven 模块（坐标），需要和项目管理者商议一下，我只是一个普通贡献者，这个做不了主

xuxiaowei-com-cn commented:
@uuuyuqi 我想问一下，代码审核就这么随意吗？看一下 https://github.com/alibaba/spring-cloud-alibaba/pull/4146 这个 PR，还有我的评论：https://github.com/alibaba/spring-cloud-alibaba/pull/4146#issuecomment-3658636543 ，看一下当前的这个 PR，

@fangjian0423 
@flystar32 
@mercyblitz 
@DanielLiu1123

uuuyuqi commented:
LGTM. Really appreciate your contribution!

xuxiaowei-com-cn commented:
1. `alibaba/spring-cloud-alibaba` 官方 GitHub Package Maven 快照已发布，建议大家使用官方快照。
2. 没有特别情况下，个人快照 https://gitlab.xuxiaowei.com.cn/xuxiaowei-com-cn/spring-cloud-alibaba/ 不再更新。

https://github.com/alibaba/spring-cloud-alibaba/issues/4185
